packages/framework/src/Component/Elasticsearch/IndexRepository.php

```diff
@@ -0,0 +1,262 @@
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\IndexDefinition $indexDefinition
+     */
+    public function deleteIndexByIndexDefinition(IndexDefinition $indexDefinition): void
+    {
+        $indexes = $this->elasticsearchClient->indices();
+        foreach ($indexes->getAliases() as $indexName => list('aliases' => $aliases)) {
+            if (array_key_exists($indexDefinition->getIndexAlias(), $aliases)) {
+                $this->deleteIndex($indexName);
+            }
+        }
+    }
+
+    /**
+     * @param string $indexName
+     */
+    public function deleteIndex(string $indexName): void
+    {
+        $indexes = $this->elasticsearchClient->indices();
+
+        $result = $indexes->delete([
+            'index' => $indexName,
+        ]);
+
+        if (isset($result['error'])) {
+            throw ElasticsearchIndexException::deleteIndexError($indexName, $result['error']);
+        }
+    }
+
+    /**
+     * @param string $indexAlias
+     * @param int $domainId
+     * @param array $ids
+     */
+    public function deleteIds(string $indexAlias, int $domainId, array $ids): void
+    {
+        $this->elasticsearchClient->deleteByQuery([
+            'index' => $indexAlias,
+            'type' => '_doc',
+            'body' => [
+                'query' => [
+                    'bool' => [
+                        'must' => [
+                            'ids' => [
+                                'values' => array_values($ids),
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ]);
+    }
+
+    /**
+     * @param string $indexAlias
+     * @param array $data
+     * @param bool $createIfNotExists
+     */
+    public function bulkUpdate(string $indexAlias, array $data, bool $createIfNotExists = true): void
+    {
+        $params = ['body' => []];
+
+        foreach ($data as $id => $row) {
+            $params['body'][] = [
+                'update' => [
+                    '_index' => $indexAlias,
+                    '_type' => '_doc',
+                    '_id' => (string)$id,
+                ],
+            ];
+
+            $params['body'][] = [
+                'doc' => $row,
+                'doc_as_upsert' => $createIfNotExists,
+            ];
+        }
+
+        $result = $this->elasticsearchClient->bulk($params);
+
+        if (isset($result['errors']) && $result['errors'] === true) {
+            throw ElasticsearchIndexException::bulkUpdateError($indexAlias, $result['items']);
+        }
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\IndexDefinition $indexDefinition
+     * @param array $restrictToIds
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     */
+    public function export(IndexDefinition $indexDefinition, array $restrictToIds, OutputInterface $output): void
+    {
+        $indexAlias = $indexDefinition->getIndexAlias();
```

Comment:
I noticed you removed disabled SQL Logging (`SqlLoggerFacade`).
This results in high memory usage on a larger data set
![Snímek obrazovky 2020-02-07 v 16 26 45](https://user-images.githubusercontent.com/1177414/74054103-984cee00-49dd-11ea-98a4-48336f9d2dc8.png)
.